packages/react-server/src/ReactFizzHooks.js

```diff
@@ -236,6 +249,18 @@ export function finishHooks(
 
     children = Component(props, refOrContext);
   }
+
+  if (bufferedChunks !== null) {
+    // This component emitted some chunks. They are buffered until it finishes
+    // rendering successfully. We can flush them now.
+    const task: Task = (currentlyRenderingTask: any);
+    const target = task.blockedSegment.chunks;
+    // Not sure why Flow complains. If I change it to spread instead of apply,
+    // it checks fine.
+    // $FlowFixMe[method-unbinding]
+    target.push.apply(target, bufferedChunks);
+  }
```

Comment:
I'm not sure this is going to work in the edge case

```
function App() {
  let ... = useFormState();
  return (
    <html>
      <body>
        <form ...>
      </body>
    </html>
}
```

In Fizz (with float enabled) the html chunks are sequester so the fact that the comment is getting pushed to the target before the DOCTYP does it accidentally works out. (we'll still emit `<DOCTYPE html><html><head>`<!--F-->`) however when hydrating on the client the starting point for hydration is `document.body` I believe (I can't remember if I made this change or I just talked about making it but conceptually we can start hydration from body b/c everything else is singleton which has it's own hydration path). If hydration starts from the body the form comment will be missed b/c it is in the head.

I don't have a good solution to this problem. We could maybe figure out if we need to put the comment before the documentElement in the DOM and slot it in between the DOCTYPE and the `<html>`. Then we'd have to also special case the hydration cursor to look there. Additionally I think this may conflict with prerendering b/c we will have a static preamble (the html, head tag, and some head content) even when we can't fully prerender the entire shell. So we may have something dynamic (the form state indictator comment) that needs to be embedded inside the static part which just can't happen

Reply:
Another crude way to do it could be to forbid useFormState outside of the body 😆